src/kademlia/node_id.cpp

```diff
@@ -70,6 +71,25 @@ int distance_exp(node_id const& n1, node_id const& n2)
 	return (std::max)(159 - distance(n1, n2).count_leading_zeroes(), 0);
 }
 
+int min_distance_exp(node_id const& n1, std::vector<node_id> const& ids)
+{
+	// specialized for cases of 0, 1 and 2 for performance reasons
+	if (ids.size() == 0) return 0;
+	if (ids.size() == 1) return distance_exp(n1, ids[0]);
+	if (ids.size() == 2)
+		return std::min(distance_exp(n1, ids[0]), distance_exp(n1, ids[1]));
```

Comment:
these special cases, is there a good reason to have them? would you imagine that they are significantly more efficient than the loop? (there's also a cost of checking the size by having them)